### DISCUSSION TITLE
Seeking help crypto trading bot

### DISCUSSION THREAD

**Discussion by Miwkasplace** [community/community#139949] [Programming Help]
### Select Topic Area

General

### Body

I’m currently working on developing a crypto trading bot for Binance, but I’ve encountered multiple challenges along the way and could really use some help. Despite using resources like ChatGPT and other tools, the bot still isn’t functioning a Here are some of the issues I’ve faced so far: Profitability Issues: The bot tends to make small gains but then quickly loses them in subsequent trades, failing to sustain profits over time. I’ve tried implementing various strategies like RSI, MACD, Bollinger Bands, and EMA, but the results are inconsistent.

High-Frequency Trading Challenges: I’ve aimed to run the bot on a high-frequency basis with multiple iterations, but I’ve had trouble optimizing the number of trades to achieve a 50-60% daily profit target. In addition, the bot often executes too many low-quality trades, leading to excessive fees and losses.

Machine Learning Integration: I’ve attempted to implement machine learning to improve the bot’s performance over time, but I’m unsure if the approach or model is correct. The learning process doesn’t seem to have a noticeable impact on the bot’s trading decisions.

API Key Management: In some cases, there have been issues with using multiple API keys for different projects, which resulted in mismanagement of permissions and failures in connecting to Binance's various services (e.g., margin trading, futures).

Strategy Optimization: Despite backtesting and live testing different strategies with simulated money, I’ve had trouble translating the backtesting results into profitable real-money trading.

Key indicators like stop-loss and take-profit values don’t seem to be optimized correctly.

Given all of these challenges, I’m wondering: Has anyone successfully developed a profitable crypto trading bot? If so, would you be willing to share your approach, strategies, or even code? Is there a recommended way to integrate advanced techniques like machine learning into a trading bot to continuously improve performance? How should I go about this process? I’d appreciate any advice, insights, or resources. I’ve been working on this for quite some time, and I’m hoping to learn from anyone who’s faced and overcome similar challenges. Thanks in advance!

I’ve been working on this for quite some time, and I’m hoping to learn from anyone who’s faced and overcome similar challenges. Thanks in advance!

**Comment by LiteBrite82** [comment]
Hi @Miwkasplace ,

Thanks for being a part of the GitHub Community, we're glad you're here!

If you're looking for help for this specific topic, you might want to try asking for help somewhere that focuses on this topic. It's possible that another GitHub user might have run into this same issue and can help, but the GitHub Community Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

In our [Community Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md), we ask that members contribute in a positive and constructive way. This type of post is considered "off topic" for our Community Discussions. Any repeat and future violations may result in a temporary or indefinite block from the Community. Thanks for understanding. 

Best of luck!
